hash_practice/exercises.py

```diff
@@ -22,8 +51,76 @@ def valid_sudoku(table):
         Each element can either be a ".", or a digit 1-9
         The same digit cannot appear twice or more in the same 
         row, column or 3x3 subgrid
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2) but I'm not sure. 
+        Space Complexity: O(1). While I create arrays, they're just temporary 
     """
-    pass
+    valid_element = {
+        '1':1, '2':1, '3':1, '4':1, '5':1,
+        '6':1, '7':1, '8':1, '9':1, '.':1
+    }
+    
+    vertical_check = True

```

Comment:
It looks like this and `horizontal_check` ended up being not used, so you can just get rid of them.